scripts/fhir_scripts.sh

```diff
@@ -204,6 +204,108 @@ function igtools_version() {
+
+    if [[ -z "$current_version" ]]; then
+        echo "not_installed"
+        return 0
+    fi
+
+    # Try to fetch latest version from GitHub with 1s timeout
+    response=$(curl -s --connect-timeout 1 --max-time 1 "$repo_url" 2>/dev/null || true)
+
+    if [[ -z "$response" ]]; then
+        echo "error:fetch_failed_or_timeout:${current_version}"
+        return 0
+    fi
+
+    # Try to find versions from filenames (dist/*.tar.gz, *.whl)
+    latest_version=$(echo "$response" \
+        | grep -Eo "${package_name}-[0-9]+(\.[0-9]+){1,2}(\.tar\.gz|\.whl)" \
+        | sort -V \
+        | tail -n 1 \
+        | grep -Eo '[0-9]+(\.[0-9]+){1,2}' || true)
+
+    # If not found, try to extract from __VERSION__ in source code listing
+    if [[ -z "$latest_version" ]]; then
+        latest_version=$(echo "$response" \
+            | grep -Eo "__VERSION__\s*=\s*(Version\()?['\"][0-9]+(\.[0-9]+){1,2}['\"]\)?" \
+            | sed -E "s/.*__VERSION__\s*=\s*(Version\()?['\"]([^'\"]+)['\"]\)?/\2/" \
+            | sort -V \
+            | tail -n 1 || true)
+    fi
+
+    if [[ -z "$latest_version" ]]; then
+        echo "error:could_not_parse_latest_version:${current_version}"
+        return 0
+    fi
+
+    # Compare installed vs. latest
+    if [[ "$current_version" == "$latest_version" ]]; then
+        echo "up_to_date:${current_version}"
+    else
+        echo "update_available:${current_version}:${latest_version}"
+    fi
+
+    return 0
+}
+
+function maintain_pytools() {
+    echo "Checking epatools version"
+    result=$(check_pytool_version epatools "https://api.github.com/repos/onyg/epa-tools/contents/dist")
+    IFS=':' read -r status current latest <<< "$result"
+
+    case "$status" in
+    update_available)
+        echo "⬆️  $package_name: $current → $latest"
+        update_pytool "git+https://github.com/onyg/epa-tools.git" "Updated epa-tools to $(epatools_version)" "Failed to update epa-tools"
+        ;;
+    up_to_date)
+        echo "✅ $package_name up to date ($current)"
+        ;;
+    error*)#
+        echo "⚠️  $package_name check failed ($current)"
+        ;;
+    not_installed)
+        echo "📦 $package_name not installed"
+        ;;
+    esac
+
+    echo "Checking igtools version"
+    result=$(check_pytool_version igtools "https://raw.githubusercontent.com/onyg/req-tooling/refs/heads/main/src/igtools/versioning.py")
+    IFS=':' read -r status current latest <<< "$result"
+
+    case "$status" in
+    update_available)
+        echo "⬆️  $package_name: $current → $latest"
+        update_pytool "git+https://github.com/onyg/req-tooling.git" "Upated reqtooling to $(igtools_version)" "Failed to update req-tooling"
+        ;;
+    up_to_date)
+        echo "✅ $package_name up to date ($current)"
+        ;;
+    error*)
+        echo "⚠️  $package_name check failed ($current)"
+        ;;
+    not_installed)
+        echo "📦 $package_name not installed"
+        ;;
+    esac
+}
```

Comment:
Wouldn't it make sense to extract the duplicated code and just call it with different parameters?

Reply:
I thought about that, too. There's always a balance to be found between code duplication and too much nesting. I think both is fine, so I will create a dedicated function for that.